test/es6/bug_issue_4635.js

```diff
@@ -0,0 +1,27 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+var tests = [
+  {
+    name: "Named function expression should have a name property even if it's being assigned to an object property",
+    body: function () {
+        var obj = {};
+        obj.a = class A { };
+        assert.areEqual('A', obj.a.name, "Since the class expression is named, it should have a name property");
+    }
+  },
+  {
+    name: "Unnamed function expression should not have a name property",
+    body: function () {
+        var obj = {};
+        obj.a = class { };
+        assert.areEqual('', obj.a.name, "Since the class expression is unnamed, it should not have a name property");
```

Comment:
Is it worth having a test that tests for `obj.a.constructor.name`? Or are they guaranteed to be the same? The code that originally brought this bug to light used `this.constructor.name`.

Reply:
I added these both for instance and via method cases. Should be redundant since should be the case of `this.constructor === obj.a`. No reason not to be a little more thorough here.